### PR TITLE
Provide a useful error message when native image fails with OOM

### LINKED ISSUE #6084: build native via docker could have more informative error messages
**Describe the bug**
Someone did the superhero workshop and kicked of native builds on windows via docker (`./mvnw package -Pnative -Dquarkus.native.container-build=true
`) for all 4-5 services exhausting his docker allocated resources.

For the user the experience was three very slow builds (10-20 minutes) and a fourth that errored with something the lines: "Docker build exited with exit code 137".

would be good if we recognized the exit code and was a bit more informative, somtehing like:

"Docker build exited with exit code 137. This indicates the build hit a memory or cpu resource limit. Ensure your docker environment have enough memory or cpu available"

That would have helped user to realize what he was actually doing by running these parallel builds.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #6084

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java [hunk 1]
```diff
@@ -343,6 +348,22 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
         }
     }
 
+    private RuntimeException imageGenerationFailed(int exitValue, List<String> command) {
+        if (exitValue == OOM_ERROR_VALUE) {
+            if (command.contains("docker") && !IS_LINUX) {
+                return new RuntimeException("Image generation failed. Exit code was " + exitValue
+                        + " which indicates an out of memory error. The most likely cause is docker not being given enough memory. Also consider increasing the xmx value for native image generation by setting the \""
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java]:
```suggestion
                        + " which indicates an out of memory error. The most likely cause is Docker not being given enough memory. Also consider increasing the Xmx value for native image generation by setting the \""
```

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java [hunk 2]
```diff
@@ -343,6 +348,22 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
...
+                        + QUARKUS_XMX_PROPERTY + "\" property");
+            } else {
+                return new RuntimeException("Image generation failed. Exit code was " + exitValue
+                        + " which indicates an out of memory error. Consider increasing the xmx value for native image generation by setting the \""
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java]:
```suggestion
                        + " which indicates an out of memory error. Consider increasing the Xmx value for native image generation by setting the \""
```

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java [hunk 3]
```diff
@@ -328,8 +331,10 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
                     errorReportLatch));
             executor.shutdown();
             errorReportLatch.await();
-            if (process.waitFor() != 0) {
-                throw new RuntimeException("Image generation failed. Exit code: " + process.exitValue());
+            int exitCode = process.waitFor();
+            if (exitCode != 0) {
+                throw imageGenerationFailed(exitCode, command);
+
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java]: Can we remove this new line?

**Review by gsmet — CHANGES_REQUESTED** [review]
This is a very useful addition, thanks!

I added some very minor comments.

**Comment by geoand** [comment]
Thanks for the review @gsmet! Feedback applied

**Review by gsmet — APPROVED** [review]

**Pull request merged**